elastic/beats
# [Filebeat] Fix fileset field namespacing

## What does this PR do?

When the fields.yml file is constructed it is done by appending files together and adding some indenting.
In the case of Filebeat, a fileset's fields.yml is appended with an indent of 8 spaces to the module's fields.yml.
This generally allows for all of the filesets fields to become children of the module.

The problem we had was that the new filesets added in #19713 expected that their fields would be root fields
(not children to the module namespace). In cases where the module already existed and had declared
a module namespace field in its fields.yml this resulted in unexpectedly namespaced fieldset fields
(e.g. microsoft.rsa.* instead of rsa.*).

The size of the x-pack/filebeat index-pattern is still large (915885 bytes), but not so large that it goes beyond the Kibana
request payload limit.

## Why is it important?

Reduces the number of fields in the Filebeat mapping and allows Kibana to generate a smaller index pattern.

## Checklist


## How to test this PR locally

- Build filebeat.
- Run `filebeat setup`
- Verify that the index pattern exists.
- Verify that you can view the index pattern.
- Verify that discover works.

## Related issues

- Fixes #19965
